joblib/my_exceptions.py

```diff
@@ -10,23 +10,22 @@
 
 class JoblibException(Exception):
     """A simple exception with an error message that you can get to."""
-
     def __init__(self, *args):
-        self.args = args
-
-    def __reduce__(self):
-        # For pickling
-        return self.__class__, self.args, {}
+        # We need to implement __init__ so that it is picked in the
+        # multiple heritance hierarchy in the class created in
+        # _mk_exception. Note: in Python 2, if you implement
+        # __init__ in your exception class you need to set .args
+        # correctly, otherwise you can dump an exception instance with
+        # pickle but not load it (at load time an empty .args will be
+        # passed to the constructor)
+        Exception.__init__(self, *args)
```

Comment:
Alright I did not understand that `super` in `Base1` would trigger the `__init__` in the sibling base class `Base2`. I think the comment should be updated to make that point explicit.